sdk/core/core-util/src/delay.ts

```diff
@@ -19,33 +19,71 @@ export interface DelayOptions {
   abortErrorMsg?: string;
 }
 
+/**
+ *
+ * @param inputs - The inputs for creating an abortable promise are the
+ *                 buildPromise function and the cleanupBeforeAbort function.
+ *                 buildPromise takes both the resolve and reject functions as
+ *                 parameters. cleanupBeforeAbort is called right before the
+ *                 promise is rejected.
+ * @returns a function that takes an optional DelayOptions parameter and returns
+ *          a promise that can be aborted.
+ * @internal
+ */
+export function createAbortablePromise<T>(inputs: {
+  buildPromise: (inputs: {
```

Comment:
tbh I like the object-first approach because it forces me to name everything so the client code becomes more readable but I agree with you we should stick with our convention if we're going to export it. Addressed in https://github.com/Azure/azure-sdk-for-js/pull/24821/commits/64ea6c224b318c12ba37a1b94c45567f983e532c.